zulipterminal/helper.py

```diff
@@ -493,3 +493,95 @@ def notify(title: str, html_text: str) -> None:
+    alignments = []
+
+    # The table cells are stored row-wise:
+    # [[row0, row0, row0],
+    # [row1, row1, row1],
+    # [row2, row2, row2]].
+    # Add +1 to count the header row as well.
+    cells = [[] for _ in range(len(rows) + 1)]  # type: List[List[str]]
+
+    # Fill up the `cells` with the header/0th row and extract alignments.
+    for header in headers:
+        alignments.append(header.get(('align'), 'left'))
+        cells[0].append(header.text)
+
+    # Fill up the `cells` with body rows.
+    for index, row in enumerate(rows, start=1):
+        for tdata in row.find_all('td'):
+            cells[index].append(tdata.text)
+    return (alignments, cells)
+
+
+def unicode_border(lcorner: str, line: str, connector: str, rcorner: str,
+                   widths: List[int], newline: bool=True) -> str:
+    """
+    Given left corner, line, connecter and right corner unicode character,
+    constructs border for markup table.
+    """
+    border = lcorner
+    for width in widths:
+        # Add +2 to the width to compensate for the extra space added around
+        # each string while styling the row by `style_row`.
+        border += (line * (width + 2)) + connector
+    border = border.rstrip(connector) + rcorner
+    return border + '\n' if newline else border
+
+
+def style_row(alignments: List[str], column_widths: List[int],
+              row: List[str]) -> str:
+    """
+    Constructs styled row strip, for markup table, using unicode characters
+    and row elements.
+    """
+    aligner = {'center': str.center, 'left': str.ljust, 'right': str.rjust}
+    row_strip = u'│ '
+    for column_num, cell in enumerate(row):
+        aligned_text = aligner[alignments[column_num]](
+            cell, column_widths[column_num]
+        )
+        row_strip += (aligned_text + u' │ ')
+    row_strip = row_strip.rstrip() + '\n'
+    return row_strip
+
+
+def render_table(table_element: Any) -> List[Union[str, Tuple[str, str]]]:
+    """
+    A helper function for rendering a markup table in the MessageBox.
+    """
+    alignments, cells = parse_html_table(table_element)
+
+    # Calculate the width required for each column.
+    column_widths = [
+        len(max(column, key=lambda string: len(string)))
+        for column in zip(*cells)
+    ]
+
+    top_border = unicode_border(u'┌', u'─', u'┬', u'┐', column_widths)
+    middle_border = unicode_border(u'├', u'─', u'┼', u'┤', column_widths)
+    bottom_border = unicode_border(u'└', u'─', u'┴', u'┘', column_widths,
+                                   newline=False)
+
+    # Construct the table, row-by-row.
+    table = []  # type: List[Union[str, Tuple[str, str]]]
+    for row_num, row in enumerate(cells):
+        styled_row = style_row(alignments, column_widths, row)
+        if row_num == 0:
+            table.append(top_border)
+            # Make the header bold.
+            table.append(('bold', styled_row))
+            table.append(middle_border)
```

Comment:
Would this be clearer as two sections, one generating the header and the other the rest (the latter being the loop)?